src/core/src/op/convert_align_types.cpp

```diff
@@ -0,0 +1,171 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/op/convert_align_types.hpp"
+
+#include "itt.hpp"
+#include "validation_util.hpp"
+
+namespace ov {
+namespace op {
+namespace {
+
+std::unordered_map<size_t, element::Type> bit_to_int{
+    {4, element::i4},
+    {8, element::i8},
+    {16, element::i16},
+    {32, element::i32},
+    {64, element::i64},
+};
+
+element::Type infer_types(const v14::ConvertAlignTypes* op) {
+    const auto lhs = op->input(0);
+    const auto rhs = op->input(1);
+    const auto promote_unsafe = op->get_promote_unsafe();
+    const auto pytorch_scalar_align = op->get_pytorch_scalar_align();
+    const auto u64_promotion_target = op->get_u64_integer_promotion_target();
+    const auto supported_types = {element::dynamic,
```

Comment:
Not sure if hardcoding unsupported types would be better idea. Like if someone would add some new type and forgot to update `unsupported types` or other rule, then types could be evaluated with unexpected results. Current implementation with hardcoded supported types only might be quite ugly, but if someone would add new type, it would require to manually extend it, providing warning that rules might need an update.

Reply:
Modified it to use switch-case and separate helper instead of table, maybe this could be a good compromise?